packages/ef-runtime-client/src/EFRuntime/EFRuntime.ts

```diff
@@ -35,54 +39,38 @@ export class EFRuntime {
   async init(
     options: {
       systemCode?: string;
-      overrides?: { [propName: string]: string };
+      overrides?: { [propName: string]: IComponentInfo };
     } = {}
   ): Promise<void> {
     this.validateOptions(options);
-
-    await Promise.all([
-      this.moduleLoader.init(),
-      this.registry.fetch(options.systemCode as string),
-    ]);
-
+    await this.registry.fetch(options.systemCode as string);
     if (options.overrides) {
       this.registry.applyOverrides(options.overrides);
     }
-
-    this.loadAll();
+    await this.loadAll();
   }
 
-  loadAll(): void {
+  async loadAll(): Promise<void> {
     const components = Object.keys(this.registry.getRegistry());
-    for (const component of components) {
+    const loadPromises = components.map((component) =>
       this.load(component).catch((error) =>
         console.error(`Error loading ${component}`, error)
-      );
-    }
+      )
+    );
+    await Promise.all(loadPromises);
```

Comment:
Maybe use `Promise.allSettled` here to detect which components are not loaded. Also, I guess the above catch statement will never run because the `load` method never call `reject`